### ISSUE TITLE
flow_images_from_directory - class_mode = NULL found 0 images

### ISSUE THREAD

**Issue by KKamojjala** [rstudio/keras#1053]
I am trying to predict the class for images present in test folder which has no sub-folders for classes.

Here is the image generation for train dataset

train_data_gen = image_data_generator(
  rescale = 1/255, #,
  validation_split = 0.3
)
# # training images
train_image_array_gen <- flow_images_from_directory(train_dir,
                                                    train_data_gen,
                                                    shuffle = TRUE,
                                                    target_size = target_size,
                                                    class_mode = "categorical",
                                                    classes = my_classes_list,
                                                    subset = "training",
                                                    seed = 2020)

# validation images
valid_image_array_gen <- flow_images_from_directory(train_dir,
                                                    train_data_gen,
                                                    shuffle = TRUE,
                                                    target_size = target_size,
                                                    class_mode = "categorical",
                                                    classes = my_classes_list,
                                                    subset = "validation",
                                                    seed = 2020)

This works just fine. 

Then, I define, compile and fit my CNN with no issues. Now, I would like to predict the class for the test images which does not have any sub-folders. So, I was using this

test_data_gen <- image_data_generator(
  rescale = 1/255
)

test_image_array_gen <- flow_images_from_directory(test_dir,
                                                   test_data_gen,
                                                   shuffle = FALSE,
                                                   target_size = target_size,
                                                   class_mode  = NULL,
                                                   seed = 2020)

This does not identify any images in my test_dir. It says "Found 0 images belonging to 0 classes"

Is this a bug or am I doing something wrong ?

My session info

latform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 17134)

Matrix products: default

Random number generation:
 RNG:     Mersenne-Twister 
 Normal:  Inversion 
 Sample:  Rounding 
 
locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252    LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                           LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] tensorflow_2.2.0 forcats_0.5.0    stringr_1.4.0    dplyr_0.8.5      purrr_0.3.4      readr_1.3.1      tidyr_1.0.3      tibble_3.0.1     ggplot2_3.3.0   
[10] tidyverse_1.3.0  keras_2.2.5.0   

loaded via a namespace (and not attached):
 [1] reticulate_1.15  tidyselect_1.1.0 splines_3.6.2    haven_2.2.0      lattice_0.20-41  colorspace_1.4-1 vctrs_0.3.0      generics_0.0.2   mgcv_1.8-31     
[10] base64enc_0.1-3  rlang_0.4.6      pillar_1.4.4     glue_1.4.1       withr_2.2.0      DBI_1.1.0        rappdirs_0.3.1   dbplyr_1.4.3     modelr_0.1.7    
[19] readxl_1.3.1     lifecycle_0.2.0  munsell_0.5.0    gtable_0.3.0     cellranger_1.1.0 rvest_0.3.5      labeling_0.3     tfruns_1.4       fansi_0.4.1     
[28] broom_0.5.6      Rcpp_1.0.4.6     backports_1.1.6  scales_1.1.1     jsonlite_1.6.1   farver_2.0.3     fs_1.4.1         digest_0.6.25    hms_0.5.3       
[37] stringi_1.4.6    grid_3.6.2       cli_2.0.2        tools_3.6.2      magrittr_1.5     crayon_1.3.4     whisker_0.4      pkgconfig_2.0.3  zeallot_0.1.0   
[46] ellipsis_0.3.0   Matrix_1.2-18    xml2_1.3.2       reprex_0.3.0     lubridate_1.7.8  assertthat_0.2.1 httr_1.4.1       rstudioapi_0.11  R6_2.4.1        
[55] nlme_3.1-147     compiler_3.6.2

**Comment by KKamojjala** [comment]
Its working now. I forgot to specify the parent directory of the test_dir as my test_dir to load the test class.
